pypeit/core/gui/edge_inspector.py

```diff
@@ -0,0 +1,321 @@
+        return np.arange(self.ntrace)[indx][np.argmin(diff)]
+        
+    def move(self, pos):
+        """
+        Move the trace nearest the cursor spatial (x) position to the cursor's
+        location.
+
+        Args:
+            pos (:obj:`list`):
+                List with x and y position of the cursor at the time of the
+                window event.
+        """
+        # Find the nearest trace
+        i = self.nearest_trace(pos[0])
+        # Get the spatial offset
+        self.offset[i] = pos[0] - self.trace_cen[self.reference_row,i]
+        # Report
+        msgs.info(f'Offsetting trace {i} by {self.offset[i]} pixels')
+        # Reset the line data
+        self.trace_plot[i].set_data((self.trace_cen[:,i] + self.offset[i], self.spec_pix))
+        # Update the plot
+        pyplot.draw()
+
+    def delete(self, pos):
+        """
+        Delete the trace nearest the cursor spatial (x) position.
+
+        Args:
+            pos (:obj:`list`):
+                List with x and y position of the cursor at the time of the
+                window event.
+        """
+        # Find the nearest trace
+        i = self.nearest_trace(pos[0])
+        # Set the remove it
+        self.remove[i] = True
+        # For now, simply set the line to be invisible
+        self.trace_plot[i].set_visible(False)
+        # Update the plot
+        pyplot.draw()
+
+    def add_trace(self, spat, side):
+        """
+        Add a new trace nearest the cursor spatial (x) position.
+
+        Args:
+            spat (:obj:`float`):
+                The spatial (x) position of the cursor at the time of the window
+                event.
+            side (:obj:`int`):
+                Edge side.  Either -1 for left or 1 for right.
+        """
+        # Append to the book-keeping vectors
+        self.ntrace += 1
+        self.good = np.append(self.good, [True])
+        self.side = np.append(self.side, [side])
+        self.add = np.append(self.add, [True])
+        self.remove = np.append(self.remove, [False])
+        self.offset = np.append(self.offset, [0.])
+
+        # Predict the spatial location of the trace
+        if self.has_pca:
+            new_trace = self.edges.predict_traces(spat, side=-1)
+        else:
+            i = self.nearest_trace(spat)
+            offset = spat - self.trace_cen[self.reference_row,i]
+            new_trace = self.trace_cen[:,i] + offset
+        # Add it
+        self.trace_cen = np.hstack((self.trace_cen, new_trace[:,None]))
+        # Plot it
+        color = self._trace_color(side)
+        self.trace_plot += [self.image_plot.axes.plot(new_trace, self.spec_pix,
+                                                      color=color, lw=2)[0]]
+        # Draw the plot
+        pyplot.draw()
+        # Report
+        msgs.info(f'Added {"right" if side > 0 else "left"} trace passing through '
+                  f'({spat:.1f}, {self.reference_row:.2f}).')
+
+    def add_left(self, pos):
```

Comment:
same here, and below for `add_right`. Can we just pass in pos[0] as the argument?

Reply:
Same response.